feat: added pokemon details

Miqqy0069 opened the pull request:
- formated dart files to 120 lines
 - added pokemon details model
 - added pokemon moves model
 - added pokemon abilities model
 - added pokemon stats model
 - added moves info model
 - added stats info model
 - added ability info model
 
pr feedback
- renamed pokemon_ability -> ability
- renamed pokemon_moves -> moves
- renamed pokemon_stats -> stats
- used pokemon name instead of pokemon ID for pokemon details


![image_720](https://user-images.githubusercontent.com/75653480/226537445-67c7bfc6-bb21-4b6b-b64a-7e450f8cd464.png)

Miqqy0069 committed:
Feat: Added App State.
- Added Home page
- Added Constant File for hard coded strings.

Miqqy0069 committed:
Fix: Deleted api client dart file
- Deleted api service dart file

Miqqy0069 committed:
feat: added pokemon_api(handler)
- added api client
- added api service

Miqqy0069 committed:
feat: added pokemon api model
- added getPokemonList

Miqqy0069 committed:
feat: added pokemon details
- formated dart files to 120 lines
- added pokemon details model
- added pokemon moves model
- added pokemon abilities model
- added pokemon stats model
- added moves info model
- added stats info model
- added ability info model

Miqqy0069 committed:
merged conflicting files

Miqqy0069 committed:
feat: added pokemon details
- formated dart files to 120 lines
- added pokemon details model
- added pokemon moves model
- added pokemon abilities model
- added pokemon stats model
- added moves info model
- added stats info model
- added ability info model

Miqqy0069 committed:
feat: added pokemon details
- deleted duplicated string constant file

Miqqy0069 committed:
feat: added pokemon details
- resolved conflicts

shawn-ffufm commented on lib/api/handler/pokemon_api.dart:
```diff
@@ -19,11 +21,18 @@ class PokemonApi {
-}
\ No newline at end of file
+
+  Future<PokemonDetails> getPokemonDetails({required int id}) async {
```
use pokemon's name instead, rather than using id 🙂

shawn-ffufm commented on lib/api/handler/pokemon_api.dart:
```diff
@@ -19,11 +21,18 @@ class PokemonApi {
+
+  Future<PokemonDetails> getPokemonDetails({required int id}) async {
+    final uri = baseUri.replace(path: '${baseUri.path}/pokemon/$id/');
+    uri;
```
do you still need this?

shawn-ffufm replied:
do you still need this?

shawn-ffufm commented on lib/api/model/pokemon_ability.dart:
```diff
@@ -0,0 +1,15 @@
+
+@freezed
+class PokemonAbility with _$PokemonAbility {
+  const factory PokemonAbility({
```
kindly remove Pokemon prefix, Ability is enough 🙂 

same goes for: moves & stats

shawn-ffufm commented on lib/api/model/pokemon_details.dart:
```diff
@@ -0,0 +1,22 @@
+@freezed
+class PokemonDetails with _$PokemonDetails {
+  const factory PokemonDetails({
+    @Default(0) @JsonKey(name: 'height') double height,
```
no need to use  -@Default- here and same goes below 🙂

shawn-ffufm commented on lib/api/model/pokemon_moves.dart:
```diff
@@ -0,0 +1,15 @@
+part 'pokemon_moves.g.dart';
+
+@freezed
+class PokemonMoves with _$PokemonMoves {
```
see related feedback above 🙂

shawn-ffufm commented on lib/api/model/pokemon_stats.dart:
```diff
@@ -0,0 +1,16 @@
+part 'pokemon_stats.g.dart';
+
+@freezed
+class PokemonStats with _$PokemonStats {
```
see related feedback above 🙂

shawn-ffufm commented on pokedex_async_redux.iml:
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
why are you still able to push some iml files? This shouldn't be here anymore 🤔

shawn-ffufm reviewed (changes requested):
@Miqqy0069 see pr feedback 🙂

Miqqy0069 committed:
feat: Added pokemon details
pr feedback
- renamed pokemon_ability -> ability
- renamed pokemon_moves -> moves
- renamed pokemon_stats -> stats
- used pokemon name instead of pokemon ID for pokemon details

Miqqy0069 committed:
Delete pokedex_async_redux.iml

Miqqy0069 committed:
feat: Added pokemon details
pr feedback
- renamed pokemon_ability -> ability
- renamed pokemon_moves -> moves
- renamed pokemon_stats -> stats
- used pokemon name instead of pokemon ID for pokemon details

Miqqy0069 committed:
Merge branch 'feature/pokemon_details' of github.com:Miqqy0069/pokedex-async-redux into feature/pokemon_details

shawn-ffufm commented on lib/api/handler/pokemon_api.dart:
```diff
@@ -19,11 +21,17 @@ class PokemonApi {
+  Future<PokemonDetails> getPokemonDetails({required String name}) async {
+    final uri = baseUri.replace(path: '${baseUri.path}/pokemon/$name/');
+    return await apiClient.dio.getUri(uri).then((response) {
+      return PokemonDetails.fromJson(jsonDecode(response.toString()));
```
why do you need to convert the response to become String? Why not just pass the response along instead? 🤔

Miqqy0069 committed:
feat: added feature pokemon details
- modified response for getpokemondetails
